### PR TITLE
metabat2: add linux-aarch64 build

### PR DESCRIPTION
**Pull request by martin-g**
Describe your pull request here

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

File: recipes/metabat2/meta.yaml [hunk 1]
```diff
@@ -21,6 +23,8 @@ requirements:
     - {{ compiler('c') }}
     - cmake
     - autoconf
+    - automake
+    - libtool
     - llvm-openmp  # [osx]
     - libgomp      # [linux]
```
Comment (martin-g) [inline recipes/metabat2/meta.yaml:29]:
```suggestion
    - libgomp      # [linux]
    - sysroot_linux-aarch64 >= 2.28 # [linux and aarch64]
```
https://github.com/samtools/htscodecs/issues/88#issuecomment-1664421461

File: recipes/metabat2/meta.yaml [hunk 2]
```diff
@@ -1,18 +1,21 @@
...
   url: https://bitbucket.org/berkeleylab/metabat/get/v{{ version }}.tar.bz2
-  sha256: c6bf3a64bbb4e8a25cd10e115b9f277d768552d0b299ad3b81d7d95762c0bfe4
+  sha256: fb46f5c3d3b0c327f62a58fceee04a55b9003276bc768040ba93c9b7913b52e2
   patches:
     - cmake.patch
+    - cmake-htslib.patch # [linux and aarch64]
 
 build:
-  number: 3
+  number: 0
```
Comment (martin-g) [inline recipes/metabat2/meta.yaml:16]:
```suggestion
  number: 0
  skip: True # [osx]
```

**Comment by martin-g** [comment]
@BiocondaBot please fetch artifacts

**Comment by martin-g** [comment]
```
mgrigorov in 🌐 euler-arm-22 in /tmp/metabat2/fff via 🅒 test123 took 6s 
❯ tar xvf metabat2-2.17-hb490ba1_0.tar.bz2                                                                                                                                                                 (test123) 
info/run_exports.json
info/test/run_test.sh
info/files
info/hash_input.json
info/recipe/build.sh
info/index.json
info/recipe/cmake.patch
info/recipe/cmake-htslib.patch
info/has_prefix
info/recipe/arm_hwcap.patch
info/recipe/meta.yaml.template
info/licenses/license.txt
info/paths.json
info/recipe/meta.yaml
info/recipe/conda_build_config.yaml
info/about.json
info/git
bin/merge_depths-DEPRECATED.pl
bin/aggregateBinDepths.pl
bin/aggregateContigOverlapsByBin.pl
bin/merge_abundances.py
bin/filter_sam_by_min_len.py
bin/runMetaBat.sh
bin/metabat
bin/metabat2
bin/metabat1
bin/contigOverlaps
bin/jgi_summarize_bam_contig_depths

mgrigorov in 🌐 euler-arm-22 in /tmp/metabat2/fff via 🅒 test123 
❯ file bin/*                                                                                                                                                                                               (test123) 
bin/aggregateBinDepths.pl:           Perl script text executable
bin/aggregateContigOverlapsByBin.pl: Perl script text executable
bin/contigOverlaps:                  ELF 64-bit LSB pie executable, ARM aarch64, version 1 (SYSV), dynamically linked, interpreter /lib/ld-linux-aarch64.so.1, for GNU/Linux 3.7.0, with debug_info, not stripped
bin/filter_sam_by_min_len.py:        Python script, ASCII text executable
bin/jgi_summarize_bam_contig_depths: ELF 64-bit LSB pie executable, ARM aarch64, version 1 (GNU/Linux), dynamically linked, interpreter /lib/ld-linux-aarch64.so.1, for GNU/Linux 3.7.0, with debug_info, not stripped
bin/merge_abundances.py:             Python script, ASCII text executable
bin/merge_depths-DEPRECATED.pl:      Perl script text executable
bin/metabat:                         symbolic link to metabat2
bin/metabat1:                        ELF 64-bit LSB pie executable, ARM aarch64, version 1 (GNU/Linux), dynamically linked, interpreter /lib/ld-linux-aarch64.so.1, for GNU/Linux 3.7.0, with debug_info, not stripped
bin/metabat2:                        ELF 64-bit LSB pie executable, ARM aarch64, version 1 (GNU/Linux), dynamically linked, interpreter /lib/ld-linux-aarch64.so.1, for GNU/Linux 3.7.0, with debug_info, not stripped
bin/runMetaBat.sh:                   Bourne-Again shell script, ASCII text executable

mgrigorov in 🌐 euler-arm-22 in /tmp/metabat2/fff via 🅒 test123 
❯ ./bin/metabat2                                                                                                                                                                                           (test123) 

MetaBAT: Metagenome Binning based on Abundance and Tetranucleotide frequency (version 2:2.17 (Bioconda); 2024-06-20T09:43:39)
by Don Kang (ddkang@lbl.gov), Feng Li, Jeff Froula, Rob Egan, and Zhong Wang (zhongwang@lbl.gov) 

Allowed options:
  -h [ --help ]                     produce help message
  -i [ --inFile ] arg               Contigs in (gzipped) fasta file format [Mandatory]
  -o [ --outFile ] arg              Base file name and path for each bin. The default output is fasta format.
                                    Use -l option to output only contig names [Mandatory].
  -a [ --abdFile ] arg              A file having mean and variance of base coverage depth (tab delimited; 
                                    the first column should be contig names, and the first row will be 
                                    considered as the header and be skipped) [Optional].
  -m [ --minContig ] arg (=2500)    Minimum size of a contig for binning (should be >=1500).
  --maxP arg (=95)                  Percentage of 'good' contigs considered for binning decided by connection
                                    among contigs. The greater, the more sensitive.
  --minS arg (=60)                  Minimum score of a edge for binning (should be between 1 and 99). The 
                                    greater, the more specific.
  --maxEdges arg (=200)             Maximum number of edges per node. The greater, the more sensitive.
  --pTNF arg (=0)                   TNF probability cutoff for building TNF graph. Use it to skip the 
                                    preparation step. (0: auto).
  --noAdd                           Turning off additional binning for lost or small contigs.
  --cvExt                           When a coverage file without variance (from third party tools) is used 
                                    instead of abdFile from jgi_summarize_bam_contig_depths.
  -x [ --minCV ] arg (=1)           Minimum mean coverage of a contig in each library for binning.
  --minCVSum arg (=1)               Minimum total effective mean coverage of a contig (sum of depth over 
                                    minCV) for binning.
  -s [ --minClsSize ] arg (=200000) Minimum size of a bin as the output.
  -t [ --numThreads ] arg (=0)      Number of threads to use (0: use all cores).
  -l [ --onlyLabel ]                Output only sequence labels as a list in a column without sequences.
  --saveCls                         Save cluster memberships as a matrix format
  --unbinned                        Generate [outFile].unbinned.fa file for unbinned contigs
  --noBinOut                        No bin output. Usually combined with --saveCls to check only contig 
                                    memberships
  --seed arg (=0)                   For exact reproducibility. (0: use random seed)
  -d [ --debug ]                    Debug output
  -v [ --verbose ]                  Verbose output


[Error!] There was no --inFile specified
[Error!] There was no --outFile specified
```
LGTM!

**Comment by martin-g** [comment]
@BiocondaBot please add label

**Review by mencian — APPROVED** [review]

**Pull request merged**